src/main/java/org/apache/nifi/components/listen/StandardListenPort.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.components.listen;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class StandardListenPort implements ListenPort {
+
+    private final int portNumber;
+    private final TransportProtocol transportProtocol;
+    private final List<String> applicationProtocols;
+
+    private StandardListenPort(final Builder builder) {
+        Objects.requireNonNull(builder.transportProtocol, "Transport protocol is required");
+        Objects.requireNonNull(builder.applicationProtocols, "Application protocols is required. Use empty list if there are no application protocols.");
+
+        this.portNumber = builder.portNumber;
+        this.transportProtocol = builder.transportProtocol;
+        this.applicationProtocols = builder.applicationProtocols;
+    }
+
+    @Override
+    public int getPortNumber() {
+        return portNumber;
+    }
+
+    @Override
+    public TransportProtocol getTransportProtocol() {
+        return transportProtocol;
+    }
+
+    @Override
+    public List<String> getApplicationProtocols() {
+        return applicationProtocols;
+    }
+
+    @Override
+    public String toString() {
+        return "StandardListenPort{" +
+            "portNumber=" + portNumber +
+            ", transportProtocol=" + transportProtocol +
+            ", applicationProtocols=" + applicationProtocols +
+            '}';
```

Comment:
Recommend using `"...".formatted()` for improved readability over the concatenation.